components/sup/src/manager/service/spec.rs

```diff
@@ -110,12 +110,25 @@ pub struct ServiceSpec {
 }
 
 impl ServiceSpec {
-    pub fn with_ident(ident: PackageIdent) -> Self {
+    pub fn new(ident: PackageIdent) -> Self {
         Self { ident,
+               group: DEFAULT_GROUP.to_string(),
+               application_environment: None,
                bldr_url: DEFAULT_BLDR_URL.to_string(),
-               ..Default::default() }
+               channel: ChannelIdent::stable(),
+               topology: Topology::default(),
+               update_strategy: UpdateStrategy::default(),
+               binds: Vec::default(),
+               binding_mode: BindingMode::Strict,
+               config_from: None,
+               desired_state: DesiredState::default(),
+               health_check_interval: HealthCheckInterval::default(),
+               svc_encrypted_password: None,
+               shutdown_timeout: None }
     }
 
+    fn default() -> Self { Self::new(PackageIdent::default()) }
```

Comment:
We could do something very noticeable like, `with_invalid_ident` because really this should only be used for deserialization, correct? What do you think?

Reply:
Maybe `deserialization_base`? Kinda drives home the "only for deserialization" aspect, which is probably good to call out.

Reply:
My take would be using the name `with_invalid_ident` (or perhaps `with_null_ident` because sadly, it's valid, just not very useful) and putting the reasoning about serialization in the doc comment. The reason being that the most importing thing for calling code to know is that this will generate a value which contains an ident that is empty. If someone wonders _why_, digging into the docs is natural.